kafka-streams-partitioners/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/RemoteServiceBasedConfigProvider.java

```diff
@@ -0,0 +1,56 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.time.Duration;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.partitioner.config.service.v1.GetPartitionerProfileRequest;
+import org.hypertrace.partitioner.config.service.v1.PartitionerConfigServiceGrpc.PartitionerConfigServiceBlockingStub;
+
+@Slf4j
+public class RemoteServiceBasedConfigProvider implements ConfigProvider {
+  private final String profileName;
+  private final LoadingCache<String, MultiLevelPartitionerConfig> partitionerConfigCache;
+
+  public RemoteServiceBasedConfigProvider(
+      PartitionerConfigServiceBlockingStub service, String profileName, Duration refreshInterval) {
```

Comment:
We want to be careful accepting a stub - that leaves each implementer responsible for building it. If it were me, I'd probably go with a config (ironic right? a config for a config provider) and also ask for a channel registry. Do we have those available where this would be constructed? So a caller could build it like

```java
new RemoteServiceBasedConfigProvider(XyzConfig.from(config.get("partitioner.config")), this.channelRegistry);
```

Something like that. We don't want to hardcode anything or build a channel that's not managed by the application's channel registry, but we also don't want to force every consumer to parse config with things like the remote host/port, profile name, duration etc.

If they wanted to manage their config locally (or for tests), consumer could always build the config by hand (`XyzConfig.builder().setHost(...)....build())`)

Reply:
Will fix it.